ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,295 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Third-party Imports
+from boto.ec2.elb.healthcheck import HealthCheck
+import boto.exception
+
+# Cloudify imports
+from ec2 import constants
+from ec2 import connection
+from ec2 import utils
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+
+@operation
+def creation_validation(**_):
+    """ This checks that all user supplied info is valid """
+
+    for property_key in constants.ELB_REQUIRED_PROPERTIES:
+        utils.validate_node_property(property_key, ctx.node.properties)
+
+    if not ctx.node.properties['resource_id']:
+        elb = None
+    else:
+        elb = _get_existing_elb(
+            ctx.node.properties['resource_id'])
+
+    if ctx.node.properties['use_external_resource'] and not elb:
+        raise NonRecoverableError(
+            'External resource, but the supplied '
+            'elb does not exist in the account.')
+
+    if not ctx.node.properties['use_external_resource'] and elb:
+        raise NonRecoverableError(
+            'Not external resource, but the supplied '
+            'elb exists.')
+
+
+@operation
+def create_elb(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        return
+
+    lb = _create_elb()
+
+    health_checks = ctx.node.properties.get('health_checks')
+
+    if isinstance(health_checks, list):
+        if health_checks:
+            for health_check in health_checks:
+                _add_health_check_to_elb(lb, health_check)
+    else:
+        raise NonRecoverableError(
+            'Health Checks not defined properly in inputs. '
+            'Health Checks should be a list.')
+
+
+def _add_instance_to_elb_list_in_properties(instance_id):
+
+    if 'instance_list' not in ctx.target.instance.runtime_properties.keys():
+        ctx.target.instance.runtime_properties['instance_list'] = []
+
+    ctx.target.instance.runtime_properties['instance_list'].append(instance_id)
+
+
+def _remove_instance_from_elb_list_in_properties(instance_id):
+    if instance_id in ctx.target.instance.runtime_properties['instance_list']:
```

Comment:
so we should check if it exists and if not, throw an error. This will just throw a weird error in which it can't find a key...